WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetailsViewModel.swift

```diff
@@ -68,14 +69,17 @@ final class CouponDetailsViewModel: ObservableObject {
 
     private let stores: StoresManager
     private let currencySettings: CurrencySettings
+    let isEditingEnabled: Bool
 
     init(coupon: Coupon,
          stores: StoresManager = ServiceLocator.stores,
-         currencySettings: CurrencySettings = ServiceLocator.currencySettings) {
+         currencySettings: CurrencySettings = ServiceLocator.currencySettings,
+         featureFlags: FeatureFlagService = ServiceLocator.featureFlagService) {
         self.siteID = coupon.siteID
         self.coupon = coupon
         self.stores = stores
         self.currencySettings = currencySettings
+        isEditingEnabled = featureFlags.isFeatureFlagEnabled(.couponEditing)
```

Comment:
While reviewing #6531 I recalled that we only enable editing coupons with supported discount types. Maybe we should also check for the discount type here and not only the feature flag?

Reply:
TIL! Updated the logic here 9a4eab10c1cdc2ffba2f9b6c51ecc1bed3c57762